src/app/support/support.component.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    Inject,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    Renderer,
+    ViewChild,
+} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {ActivatedRoute, Router} from '@angular/router';
+import {SupportRestService} from '../rest/support.service';
+import {Locale} from "../locale/locale";
+import {
+    MatDialog,
+    MatPaginatorIntl,
+    MatSort,
+    MatTableDataSource,
+    PageEvent
+} from '@angular/material';
+import { NgForm } from '@angular/forms';
+import "rxjs/add/operator/toPromise";
+import {
+    SupportRequest
+} from './support.definitions';
+
+declare var $: any;
+declare var Chartist: any;
+declare var swal: any;
+declare var classie: any;
+
+declare function require(name: string);
+
+
+@Component({
+  selector: 'app-support',
+  templateUrl: './support.component.html',
+  styleUrls: ['./support.component.css']
+})
+
+
+export class SupportComponent implements OnInit, OnDestroy, AfterViewInit {
+
+    public timerId: any;
+    public userName : string;
+    public userEmail : string;
+    public title : string;
+    public description : string;
+    public sendSystemInfo : boolean;
+
+    constructor(private http: HttpClient, 
+    			private route: ActivatedRoute,
+                private supportRestService: SupportRestService,
+                private renderer: Renderer,
+                public router: Router,
+                public dialog: MatDialog,
+                private cdr: ChangeDetectorRef,
+                private matpage: MatPaginatorIntl,
+    ) {
+
+    }
+
+    ngOnInit() {
+
+    }
+
+    ngAfterViewInit() 
+    {
+
+    }
+
+    ngOnDestroy() {
+    }
+    
+    sendRequest(requestForm: NgForm): void {
+        if (!requestForm.valid) {
+            return;
+        }
+
+        var request = new SupportRequest();
+        
+        request.name = this.userName;
+        request.email = this.userEmail;
+        request.title = this.title;
+        request.description = this.description;
+        request.sendSystemInfo = this.sendSystemInfo;
+        
+        this.supportRestService.sendRequest(request).subscribe(data => {
+            if (data["success"] == true) {
+                $.notify({
+                    icon: "ti-email",
+                    message: "Request Sent"
```

Comment:
Giving more friendly messages would be better for the success and failure cases

"Your Support request has been sent successfully. "

"Your request has been sent. Support team will contact through your e-mail soon."

Reply:
Absolutely